pyiceberg/table/__init__.py

```diff
@@ -2010,6 +2016,84 @@ def create_branch(
         self._requirements += requirement
         return self
 
+    def rollback_to_snapshot(self, snapshot_id: int) -> ManageSnapshots:
+        """Rollback the table to the given snapshot id.
+
+         The snapshot needs to be an ancestor of the current table state.
+
+        Args:
+            snapshot_id (int): rollback to this snapshot_id that used to be current.
+        Returns:
+            This for method chaining
+        """
+        self._commit_if_ref_updates_exist()
+        if self._transaction._table.snapshot_by_id(snapshot_id) is None:
+            raise ValidationError(f"Cannot roll back to unknown snapshot id: {snapshot_id}")
+        if snapshot_id not in {
+            ancestor.snapshot_id
+            for ancestor in ancestors_of(self._transaction._table.current_snapshot(), self._transaction.table_metadata)
+        }:
+            raise ValidationError(f"Cannot roll back to snapshot, not an ancestor of the current state: {snapshot_id}")
+
+        update, requirement = self._transaction._set_ref_snapshot(snapshot_id=snapshot_id, ref_name="main", type="branch")
+        self._updates += update
+        self._requirements += requirement
```

Comment:
It seems this pattern is and will be required for every API in `ManageSnapshot`. Shall we extract it to a helper to reduce code duplication?

We could also consider move `_set_ref_snapshot` into `ManageSnapshot` if it helps, since that is used only in this class.

Reply:
I've added a helper function `_stage_main_branch_snapshot_ref`. I feel like it can be renamed to describe the operation better.

I'll also move the `_set_ref_snapshot` function to `ManageSnapshot`